controllers/pvc-mutator/encryption/keygen.go

```diff
@@ -101,21 +101,29 @@ func (s *EncryptionKeySetter) MutatePVC(ctx context.Context, pvc *corev1.Persist
 	log := s.log.WithValues("pvc", client.ObjectKeyFromObject(pvc).String())
 	log.V(4).Info("received pvc for mutation")
 
-	// Skip mutation if the PVC is not provisioned by StorageOS
-	provisioned, err := provisioner.IsProvisionedPVC(s.Client, *pvc, namespace, provisioner.DriverName)
+	// Retrieve StorageClass of the PVC.
+	storageClass, err := provisioner.StorageClassForPVC(s.Client, pvc)
 	if err != nil {
-		return errors.Wrap(err, "failed to check pvc provisioner")
+		return errors.Wrap(err, "failed to retrieve storageclass of pvc")
 	}
+
+	// Skip mutation if the PVC is not provisioned by StorageOS
+	provisioned := provisioner.IsProvisionedStorageClass(storageClass, provisioner.DriverName)
 	if !provisioned {
 		log.V(4).Info("pvc does not provisioned by StorageOS, skipping")
 		return nil
 	}
 
-	// Skip mutation if the PVC does not have encryption enabled.  Don't bother checking
-	// the StorageClass to make sure it's StorageOS.  The encryption label
-	// should only be added to StorageOS PVCs.
-	if !isEnabled(s.enabledLabel, pvc.GetLabels()) {
-		log.V(4).Info(fmt.Sprintf("pvc does not have %s=true annotation, skipping", s.enabledLabel))
+	// Skip mutation if the PVC does not have encryption enabled.
+	// The encryption label should be added to StorageOS PVCs
+	// or inherited from StorageOS StorageClass.
+	// Invalid value of encryption must block PVC creation.
+	enabled, err := s.isEnabled(pvc.GetLabels(), storageClass.Parameters)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to parse boolean value of %s pvc", s.enabledLabel))
```

Comment:
This would render something like:
```
failed to parse boolean value of storageos/encryption pvc
```

Maybe the message could be improved?
Also, I'm not sure if as a caller of `isEnabled()`, the error here should mention about boolean parsing. The caller need not be specific about how the result is determined in that function. Just a thought.

Reply:
But somehow the caller has to realize the error, because forcing a value on case of parse error should lead different behavior than the user expected. Plain volume instead of encrypted or the opposite.

Reply:
Maybe just `failed to parse "storageos/encryption" option on PVC` should be good. The wrapped error will have the details.

Reply:
`failed to parse "storageos/encryption" option for PVC` because we parse both PVC and storageclass?

Reply:
oh right! Maybe volume instead of PVC would be better?

Reply:
oh I see. `for`. Sounds right to me.

Reply:
How about: `failed to parse boolean value for "storageos.com/encryption" pvc label or storageclass parameter`?  I think that covers both, and explains what's expected.